src/main/java/com/uber/cadence/internal/sync/WorkflowClientInternal.java

```diff
@@ -71,6 +71,8 @@ public static WorkflowClient newInstance(
     Objects.requireNonNull(options);
 
     emitClientVersion(options);
+    WorkerShutDownHandler.registerHandler();
```

Comment:
This seems like a weird place to register worker shutdown handler, because the WorkflowClient can also be used for non-worker purposes (ie. starting/signaling a workflow from any Java service). 

How about doing the registration in the WorkerFactory, ie. WorkerFactor.newInstance()?